src/main/java/com/example/demo/OrderComponent/Domain/Services/OrderService.java

```diff
@@ -0,0 +1,214 @@
+package com.example.demo.OrderComponent.Domain.Services;
+
+import com.example.demo.helper.enums.Currency;
+import com.example.demo.OrderComponent.API.DTOs.*;
+import com.example.demo.OrderComponent.Domain.Entities.Order;
+import com.example.demo.OrderComponent.Domain.Entities.OrderItem;
+import com.example.demo.OrderComponent.Domain.Entities.Enums.OrderStatus;
+import com.example.demo.OrderComponent.Domain.Entities.OrderItemModifier;
+import com.example.demo.OrderComponent.Helpers.Mappers.OrderMapper;
+import com.example.demo.OrderComponent.Helpers.OrderHelper;
+import com.example.demo.OrderComponent.Repositories.IOrderItemModifierRepository;
+import com.example.demo.OrderComponent.Repositories.IOrderRepository;
+import com.example.demo.OrderComponent.Repositories.IOrderItemRepository;
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class OrderService {
+    private final IOrderRepository orderRepository;
+    private final IOrderItemRepository orderItemRepository;
+    private final IOrderItemModifierRepository orderItemModifierRepository;
+
+    public OrderService(IOrderRepository orderRepository, IOrderItemRepository orderItemRepository, IOrderItemModifierRepository orderItemModifierRepository) {
+        this.orderRepository = orderRepository;
+        this.orderItemRepository = orderItemRepository;
+        this.orderItemModifierRepository = orderItemModifierRepository;
+    }
+
+    public OrderDTO createOrder(OrderDTO createOrderDTO) {
+        Order order = new Order();
+        order.setBusinessId(createOrderDTO.getBusinessId());
+        order.setCreatedByEmployeeId(createOrderDTO.getEmployeeId());
+        order.setCreatedAt(LocalDateTime.now());
+        order.setStatus(OrderStatus.NEW);
+        order.setReservationId(createOrderDTO.getReservationId());
```

Comment:
New entity = use builder (most likely put to mapper too, don't create entities on services, that's what mappers are used for)